packages/flutter_tools/test/src/common.dart

```diff
@@ -145,6 +145,19 @@ Future<void> expectToolExitLater(Future<dynamic> future, Matcher messageMatcher)
   }
 }
 
+Matcher containsIgnoreWhitespacesAndLineBreaks(String toContain){
+return  predicate((String value) => _ignoreWhitespace(_ignoreLinebreak(value))
+    .contains(_ignoreWhitespace(_ignoreLinebreak(toContain))),
```

Comment:
Rather than having to seperately handle newlines and spaces (and to additionally cover tabs), you can instead do:

`.contains(toContain.replaceAll(RegExp(r'\s'), '')),`

And you can shorten the name of this function to `containsIgnoreWhitespace`, as I think newlines are usually understood to be a subset of whitespace.

Reply:
@kwkr ^

And it also looks like you have trailing spaces that the analyzer is catching:

```
/tmp/flutter sdk/packages/flutter_tools/test/general.shard/channel_test.dart:295: trailing U+0020 space character
/tmp/flutter sdk/packages/flutter_tools/test/general.shard/commands/build_appbundle_test.dart:414: trailing U+0020 space character
/tmp/flutter sdk/packages/flutter_tools/test/general.shard/commands/build_appbundle_test.dart:416: trailing U+0020 space character
```

Reply:
How about now?

Also how is it possible that my analyses claim "No issues found!"? I followed:
https://github.com/flutter/flutter/wiki/Using-the-Dart-analyzer
Is it up to date?

Reply:
Hmm, not sure. I know that the `analyze-linux` runs additional tests that a mere `flutter analyze` won't do, so the trailing whitespace check might be a custom test we wrote (it seems that should be an analyzer lint though...)

Reply:
Yeah, that is indeed what is happening: https://github.com/flutter/flutter/blob/0d86bfc07687af2069d1d09507235934cdaba05c/dev/bots/analyze.dart#L599

Reply:
If you want to run this locally (I seldom do, only if I see something fail on CI and don't understand), you can run from your repo root:

`./bin/cache/dart-sdk/bin/dart --enable-asserts ./dev/bots/analyze.dart`